src/credentials/examples/DeviceAttestationVerifierExample.cpp

```diff
@@ -0,0 +1,194 @@
+                0x16, 0x80, 0x14, 0xEF, 0x18, 0xE0, 0xEC, 0xD4, 0x66, 0x04, 0x34, 0xDF, 0x0D, 0xBC, 0x91, 0x1E, 0xD4, 0x52, 0x16,
+                0x99, 0x66, 0x83, 0x9F, 0x30, 0x0A, 0x06, 0x08, 0x2A, 0x86, 0x48, 0xCE, 0x3D, 0x04, 0x03, 0x02, 0x03, 0x48, 0x00,
+                0x30, 0x45, 0x02, 0x21, 0x00, 0xA7, 0xFA, 0x5F, 0xA7, 0x66, 0xFF, 0xF7, 0xB5, 0xFD, 0x15, 0x8C, 0xD3, 0x4F, 0x62,
+                0x5D, 0x48, 0x91, 0x20, 0x0E, 0x05, 0x85, 0x0E, 0xF1, 0x25, 0xF6, 0xDE, 0xE7, 0x56, 0x54, 0xF9, 0x97, 0x36, 0x02,
+                0x20, 0x23, 0xB5, 0x1B, 0x6E, 0x42, 0x53, 0x28, 0x28, 0xDB, 0x09, 0x92, 0x03, 0x37, 0x05, 0x0D, 0x70, 0xF0, 0xB6,
+                0xB7, 0x2C, 0xCD, 0x3C, 0xB7, 0x81, 0xE6, 0x71, 0xAC, 0x22, 0x5D, 0xF5, 0x47, 0x2B },
+              413,
+              { 0xEF, 0x18, 0xE0, 0xEC, 0xD4, 0x66, 0x04, 0x34, 0xDF, 0x0D,
+                0xBC, 0x91, 0x1E, 0xD4, 0x52, 0x16, 0x99, 0x66, 0x83, 0x9F } }
+        };
+
+    size_t paaLookupTableIdx;
+    for (paaLookupTableIdx = 0; paaLookupTableIdx < sizeof(sPAALookupTable) / sizeof(*sPAALookupTable); ++paaLookupTableIdx)
+    {
+        if (skid.data_equal(ByteSpan(sPAALookupTable[paaLookupTableIdx].mSKID)))
+        {
+            break;
+        }
+    }
+
+    VerifyOrReturnError(paaLookupTableIdx < sizeof(sPAALookupTable) / sizeof(*sPAALookupTable), CHIP_ERROR_INVALID_ARGUMENT);
+
+    return CopySpanToMutableSpan(ByteSpan{ sPAALookupTable[paaLookupTableIdx].mPAACertificate }, out_dac_buffer);
+}
+
+class ExampleDACVerifier : public DeviceAttestationVerifier
+{
+public:
+    AttestationVerificationResult
+    VerifyAttestationInformation(const ByteSpan & attestation_info_buffer, const ByteSpan & attestation_challenge_buffer,
+                                 const ByteSpan & attestation_signature_buffer, const ByteSpan & pai_cert_der_buffer,
+                                 const ByteSpan & dac_cert_der_buffer, const ByteSpan & attestation_nonce) override;
+};
+
+AttestationVerificationResult ExampleDACVerifier::VerifyAttestationInformation(const ByteSpan & attestation_info_buffer,
+                                                                               const ByteSpan & attestation_challenge_buffer,
+                                                                               const ByteSpan & attestation_signature_buffer,
+                                                                               const ByteSpan & pai_cert_der_buffer,
+                                                                               const ByteSpan & dac_cert_der_buffer,
+                                                                               const ByteSpan & attestation_nonce)
+{
+    // match DAC and PAI VIDs
+    if (!pai_cert_der_buffer.empty())
+    {
+        VendorId paiVid;
+        VendorId dacVid;
+
+        VerifyOrReturnError(ExtractVIDFromX509Cert(pai_cert_der_buffer, paiVid) == CHIP_NO_ERROR,
```

Comment:
VID is not mandatory in PAI. Should use CHIP_ERROR_NOT_FOUND to indicate lack of VID, but if present, it should be cached for the subsequent verifications.

Reply:
will update